gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcApiExceptionFactory.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.api.core.BetaApi;
+import com.google.api.gax.rpc.AbortedApiException;
+import com.google.api.gax.rpc.AlreadyExistsApiException;
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.CancelledApiException;
+import com.google.api.gax.rpc.DataLossApiException;
+import com.google.api.gax.rpc.DeadlineExceededApiException;
+import com.google.api.gax.rpc.FailedPreconditionApiException;
+import com.google.api.gax.rpc.InternalApiException;
+import com.google.api.gax.rpc.InvalidArgumentApiException;
+import com.google.api.gax.rpc.NotFoundApiException;
+import com.google.api.gax.rpc.OutOfRangeApiException;
+import com.google.api.gax.rpc.PermissionDeniedApiException;
+import com.google.api.gax.rpc.ResourceExhaustedApiException;
+import com.google.api.gax.rpc.UnauthenticatedApiException;
+import com.google.api.gax.rpc.UnavailableApiException;
+import com.google.api.gax.rpc.UnknownApiException;
```

Comment:
After seeing the exception names, I think `Api` should be removed from all of them (except the parent, `ApiException`).